base/sort.jl

```diff
@@ -367,6 +409,40 @@ end
 
 sort(v::AbstractVector; kws...) = sort!(copy(v); kws...)
 
+
+## selectperm: the permutation to sort the first k elements of an array ##
+
+function selectperm(v::AbstractVector,
+                    k::Union(Int,OrdinalRange);
+                    lt::Function=isless,
+                    by::Function=identity,
+                    rev::Bool=false,
+                    order::Ordering=Base.Order.Forward)
+    select!(collect(1:length(v)), k, Perm(ord(lt, by, rev, order), v))
```

Comment:
It would be more standard (and simpler) if this function were to call `selectperm!`.  E.g., 

```julia
selectperm(v::AbstractVector, k::Union(Integer,OrdinalRange); kwargs...) = 
    selectperm!(Vector{eltype(k)}(length(v)), v, k, kwargs...)
```